new-docs/content/any/getting-started/policies/index.md

```diff
@@ -9,325 +9,438 @@ description: Learn about writing Oso policies - the source of truth for authoriz
+* `Float`
 
-Either way, using the rule could look like this:
+These classes correspond to ones in the application language,
+e.g., in Python, `String` is actually the built-in `str` class.
 
-{{% exampleGet "specializedExample" %}}
+## Method Calls
 
-{{% callout "Note" "green" %}}
-Type specializers automatically respect the **inheritance** hierarchy of
-application classes. See the [Resources with
-Inheritance](guides/more/inheritance) guide for an in-depth
-example of how this works.
-{{% /callout %}}
+In addition to accessing the fields of application objects in a rule,
+you can also call methods on them. We'll demonstrate this using the
+built-in `String` class, but it applies to instances of any registered
+class.
 
-Once a class is registered, class or static methods can also be called from Oso
-policies:
+Suppose we wanted to match a range of actions that all started
+with the letter "r". (A silly encoding, perhaps, but illustrative.)
+We could allow all such actions using a rule like this:
 
-{{< code file="policy.polar" >}}
-allow(actor: User, action, resource) if actor.name in User.superusers();
-{{< /code >}}
+```polar
+allow(actor, action: String, resource) if
+    action.startswith("r");
+```
 
-{{% exampleGet "classMethodExample" %}}
+This calls the {{< lang >}} method `str.startswith`, guarding
+against invalid calls with the `String` specializer.
 
-### Built-in Types
+## Other Rules
 
-Methods called on the Polar built-in types `String`, `Dictionary`, `Number`,
-and `List` punt to methods on the corresponding application language class.
-That way you can use familiar methods like `.{{% exampleGet "startswith" %}}()`
-on strings regardless of whether they originated in your application or as a
-literal in your policy. This applies to all of Polar's [supported
-types](polar-syntax#primitive-types) in any supported application language. For
-examples using built-in types, see [the {{% exampleGet "langName" %}}
-library](reference/polar/classes) guide.
+So far all of our examples have involved only `allow` rules
+with three parameters. That's a natural place to start with Oso,
+because the `is_allowed` function generates queries of the form
+`allow(actor, action, resource)`. But you can write rules for
+whatever you like, and use them from within your `allow` rules.
+For instance, the built-in [roles library](/guides/roles) supplies an
+`allow` rule that looks like this:
 
-{{% callout "Warning" "orange" %}}
-Do not attempt to mutate a literal using a method on it. Literals in Polar
-are constant, and any changes made to such objects by calling a method will
-not be persisted.
-{{% /callout %}}
+```polar
+allow(user, action, resource) if
+    resource_role_applies_to(resource, role_resource) and
+    user_in_role(user, role, role_resource) and
+    role_allow(role, action, resource);
+```
 
+Because all three parameters are variables, any three arguments
+will match the head, by binding. But queries for each of the body
```

Comment:
This almost makes me want to have a subsection under the previous "Variables" section dedicated to binding, so that if the reader already forgot what "binding" means we could send them back there. Interested if you think this would be helpful.